js/ui/date_box/ui.date_box.base.js

```diff
@@ -660,7 +648,8 @@ const DateBox = DropDownEditor.inherit({
                 if(this.option('isValid')) {
                     this._applyInternalValidation(this.dateOption('value'));
                 } else {
-                    this._validateValue(this.dateOption('value'));
+                    this._applyInternalValidation(this.dateOption('value'));
```

Comment:
The `_applyInternalValidation` method is called in both blocks now. Let's refactor this.

Reply:
fixed